src/Features/LanguageServer/ProtocolUnitTests/Completion/CompletionResolveTests.cs

```diff
@@ -106,11 +116,15 @@ class B : A
             var tags = new string[] { "Method", "Public" };
             var completionParams = CreateCompletionParams(
                 locations["caret"].Single(), LSP.VSCompletionInvokeKind.Explicit, "\0", LSP.CompletionTriggerKind.Invoked);
+            var completionList = await RunGetCompletionsAsync(testLspServer, completionParams);
+            var serverCompletionItem = completionList.Items.FirstOrDefault(item => item.Label == "M()");
+            var completionResultId = ((CompletionResolveData)serverCompletionItem.Data).ResultId.Value;
             var document = testLspServer.GetCurrentSolution().Projects.First().Documents.First();
 
-            var completionItem = await CreateCompletionItemAsync(
-                label: "M()", LSP.CompletionItemKind.Method, tags, completionParams, document,
-                commitCharacters: CompletionRules.Default.DefaultCommitCharacters).ConfigureAwait(false);
+            // Reconstruct the completion item in our view like a client would while maintaining data
+            var clientCompletionItem = await CreateCompletionItemAsync(
+                label: "M()", kind: LSP.CompletionItemKind.Method, tags: tags, request: completionParams, document: document,
```

Comment:
could we extract this out to a common helper?  Looks like this mostly similar code is repeated above

Reply:
Created a `ConvertToClientCompletionItem` method that serializes then deserializes the server item to more accurately replicate what happens in practice